DataConnectors/GoogleWorkspaceReports/GWorkspaceReportsAPISentinelConnector/__init__.py

```diff
@@ -0,0 +1,148 @@
+    next_page_token = results.get('nextPageToken', None)
+    result = results.get('items', [])
+    result_activities.extend(result)
+    while next_page_token is not None:
+        results = service.activities().list(userKey='all', applicationName=activity,
+                                            maxResults=1000, startTime=start_time, endTime=end_time, pageToken=next_page_token).execute()
+        next_page_token = results.get('nextPageToken', None)
+        result = results.get('items', [])
+        result_activities.extend(result)
+    if result_activities == None or len(result_activities) == 0:
+        print("Logs not founded for {} activity".format(activity))
+        logging.info("Logs not founded for {} activity".format(activity))
+        logging.info("Activity - {}, {} events was processed)".format(activity, len(result_activities)))
+    else:
+        logging.info("Activity - {}, {} events was processed)".format(activity, len(result_activities)))
+        return result_activities
+
+def build_signature(customer_id, shared_key, date, content_length, method, content_type, resource):
+    x_headers = 'x-ms-date:' + date
+    string_to_hash = method + "\n" + str(content_length) + "\n" + content_type + "\n" + x_headers + "\n" + resource
+    bytes_to_hash = bytes(string_to_hash, encoding="utf-8")
+    decoded_key = base64.b64decode(shared_key)
+    encoded_hash = base64.b64encode(hmac.new(decoded_key, bytes_to_hash, digestmod=hashlib.sha256).digest()).decode()
+    authorization = "SharedKey {}:{}".format(customer_id,encoded_hash)
+    return authorization
+
+def post_data(customer_id, shared_key, body, log_type):
+    method = 'POST'
+    content_type = 'application/json'
+    resource = '/api/logs'
+    rfc1123date = datetime.datetime.utcnow().strftime('%a, %d %b %Y %H:%M:%S GMT')
+    content_length = len(body)
+    signature = build_signature(customer_id, shared_key, rfc1123date, content_length, method, content_type, resource)
+    uri = 'https://' + customer_id + '.ods.opinsights.azure.com' + resource + '?api-version=2016-04-01'
+
+    headers = {
+        'content-type': content_type,
+        'Authorization': signature,
+        'Log-Type': log_type,
+        'x-ms-date': rfc1123date
+    }
+    response = requests.post(uri,data=body, headers=headers)
+    if (response.status_code >= 200 and response.status_code <= 299):
+        logging.info("Logs with {} activity was processed into Azure".format(log_type))
+    else:
+        logging.warn("Response code: {}".format(response.status_code))
+
+def expand_data(obj):
+    for event in obj:
+        for nested in event["events"]:
+            if 'name' in nested:
+                event.update({'event_name': nested["name"]})
+            if 'type' in nested:
+                event.update({'event_type': nested["type"]})
+            if 'parameters' in nested:
+                for parameter in nested["parameters"]:
+                    if 'value' in parameter:

```

Comment:
According to the API, name is also part of the nested parameter: https://developers.google.com/admin-sdk/reports/v1/reference/activities/list

Reply:
Added 'name' checking.